Aplib.Tests/Core/Belief/BeliefsetTests.cs

```diff
@@ -1,40 +1,143 @@
 ﻿using Aplib.Core.Belief;
 
 namespace Aplib.Tests.Core.Belief;
+
+/// <summary>
+/// Describes a set of tests for the <see cref="BeliefSet"/> class.
+/// </summary>
 public class BeliefSetTests
 {
     /// <summary>
-    /// Given a BeliefSet instance with multiple beliefs,
+    /// Given a BeliefSet instance with multiple <i>public field</i> beliefs,
     /// When UpdateBeliefs is called,
     /// Then all beliefs are updated.
     /// </summary>
     [Fact]
-    public void UpdateBeliefs_Called_UpdatesAllBeliefs()
+    public void UpdateBeliefs_PublicBeliefFields_UpdatesAllBeliefs()
     {
         // Arrange
-        TestBeliefSet beliefSet = new();
+        TestBeliefSetPublic beliefSet = new();
 
         // Act
+        // UpdateBeliefs should set Updated to true for all beliefs.
         beliefSet.UpdateBeliefs();
 
         // Assert
         Assert.True(beliefSet.Belief1.Updated);
         Assert.True(beliefSet.Belief2.Updated);
     }
 
-    private class TestBeliefSet : BeliefSet
+    /// <summary>
+    /// Given a BeliefSet instance with multiple <i>public property</i> beliefs,
+    /// When UpdateBeliefs is called,
+    /// Then all beliefs are updated.
+    /// </summary>
+    [Fact]
+    public void UpdateBeliefs_PublicBeliefProperties_DoesNotUpdateAnyBeliefs()
+    {
+        // Arrange
+        TestBeliefSetProperties beliefSet = new();
+
+        // Act
+        // UpdateBeliefs should *not* set Updated to true for any belief.
+        beliefSet.UpdateBeliefs();
+
+        // Assert
+        Assert.False(beliefSet.Belief1.Updated);
+        Assert.False(beliefSet.Belief2.Updated);
+    }
+
+    /// <summary>
+    /// Given a BeliefSet instance with multiple <i>private field</i> beliefs,
+    /// When UpdateBeliefs is called,
+    /// Then all beliefs are updated.
```

Comment:
It should say `Then no fields are updated.`

Reply:
Good eye!